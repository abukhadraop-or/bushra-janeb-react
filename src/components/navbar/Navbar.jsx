import { faBars, faTimes } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import React, { useState } from 'react';
import {
  Anchor,
  Links,
  LinksListItem,
  Menu,
  Navbar,
  NavbarContainer,
  TitleLink,
} from './navbar.styles';

/**
 * Render a <Navbar> component.
 *
 * @param {Object} props       Props.
 * @param {Array}  props.links The navbar links passed from the app.
 *
 * @return {JSX.Element}
 */
export const Nav = () => {
  const [isNavOpen, setIsNavOpen] = useState(false);
  const links = [
    { name: 'Home', to: '/home', id: 1 },
    { name: 'SignIn', to: '/signUp', id: 2 },
    { name: 'SignUp', to: '/login', id: 3 },
  ];

  /**
   * Handles hamburger menu click event.
   *
   */
  const handleNavClick = () => {
    setIsNavOpen(!isNavOpen);
  };

  return (
    <NavbarContainer>
      <Navbar>
        <div>
          <TitleLink>Conduit </TitleLink>
        </div>
        <Links isNavOpen={isNavOpen}>
          {links.map((link) => (
            <LinksListItem key={link.id}>
              <Anchor>{link.name}</Anchor>
            </LinksListItem>
          ))}
        </Links>
        <Menu>
          <FontAwesomeIcon
            icon={isNavOpen ? faTimes : faBars}
            label={isNavOpen ? 'bars' : 'closed'}
            onClick={() => {
              handleNavClick();
            }}
          />
        </Menu>
      </Navbar>
    </NavbarContainer>
  );
};

export default Nav;
