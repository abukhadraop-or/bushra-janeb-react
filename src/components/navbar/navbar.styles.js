import styled from 'styled-components';
import breakpoints from 'styles/breaking-points';
import colors from 'styles/global-colors';

export const NavbarContainer = styled.div`
  margin: 0 auto;
  max-width: 62rem;
  padding: 0 1rem;
`;

export const Navbar = styled.nav`
  align-items: center;
  display: flex;
  justify-content: space-between;
  padding: 0.188rem 0;
`;

export const Menu = styled.div`
  cursor: pointer;
  font-size: 1.563rem;

  @media screen and (min-width: ${breakpoints.desktop}) {
    display: none;
  }
`;

export const Links = styled.ul`
  background-color: ${colors.concrete};
  color: ${colors.black};
  display: ${({ isNavOpen }) => (isNavOpen ? 'static' : 'none')};
  font-family: 'Source Sans Pro', sans-serif;
  height: 38.2%;
  min-width: 90%;
  padding: 0;
  position: absolute;
  top: 5%;
  left: 50%;
  transform: translate(-50%, 0);
  transition: 0.5s;

  @media screen and (min-width: ${breakpoints.mobileWide}) {
    height: 33.2%;
  }

  @media screen and (min-width: ${breakpoints.desktop}) {
    background-color: ${colors.white};
    border-radius: 0;
    display: inline-flex;
    justify-content: flex-end;
    left: 0rem;
    list-style: none;
    margin: 0;
    position: relative;
    top: 0%;
    transform: none;
    transition: none;
  }
`;

export const LinksListItem = styled.li`
  display: block;
  padding-left: 1.563rem;
  padding-top: 2.188rem;
  text-align: center;

  @media screen and (min-width: ${breakpoints.desktop}) {
    padding-top: 0;
  }
`;

export const Anchor = styled.a`
  color: ${colors.midnightBlue};
  cursor: pointer;
  font-size: 0.99rem;
  text-decoration: none;

  &:hover {
    color: ${colors.Gray90};
  }
`;

export const TitleLink = styled.a`
  color: ${colors.fern};
  cursor: pointer;
  font: 1.5rem 'Titillium Web', sans-serif;
`;
