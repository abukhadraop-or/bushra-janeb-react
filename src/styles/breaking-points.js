const breakpoints = {
  // 360px
  mobile: '20rem',

  // 480px
  mobileWide: '21.438rem',

  // 650px
  tablet: '54.625rem',

  // 768px
  tabletWide: '48rem',

  // 980px
  desktop: '60rem',

  // 1180px
  desktopWide: '73.75rem',

  // 10444px
  desktopMax: '65.25',

  // 1092x
  desktopExtraWide: '68.25rem',

  // 1180px.
  LargeScreen: '73.75rem',
};

export default breakpoints;
