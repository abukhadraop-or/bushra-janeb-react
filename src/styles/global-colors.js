const colors = {
  black: '#000',
  concrete: '#f3f3f3',
  darkBlue: '#00000066',
  fern: '#5cb85c',
  Gray90: '#E5E5E5',
  lightGrey: '#bbbbbb',
  midnightBlue: '#0000004d',
  transparent: '#ffffff00',
  white: '#ffffff',
  veryLightGrey: '#ddd',
  aliceBlue: '#eceeef',
  aluminium: '#818a91',
};

export default colors;
