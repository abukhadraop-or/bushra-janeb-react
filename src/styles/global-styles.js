import { createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`
  body {
    margin: 0;
    padding: 0;
    font-family: Open-Sans, Helvetica, Sans-Serif;
  }

  *,
  *::before,
  *::after {
    box-sizing: border-box;
  }
`;

export default GlobalStyle;
