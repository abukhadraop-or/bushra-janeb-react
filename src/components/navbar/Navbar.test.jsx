import { faBars, faTimes } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Enzyme, { shallow } from 'enzyme';
import React from 'react';
import { Nav } from './Navbar';
import { Links, NavbarContainer } from './navbar.styles';

Enzyme.configure({ adapter: new Adapter() });

describe('Navbar', () => {
  const wrapper = shallow(<Nav />);
  it('should render the Navbar', () => {
    const holder = wrapper.find(NavbarContainer);
    expect(holder.exists()).toBe(true);
  });

  it('Menu should be bars', () => {
    const menu = wrapper.find(FontAwesomeIcon);
    const click = menu.simulate('click');
    expect(click.prop('icon')).toBe(faBars);
  });

  it('should change the bars into a cross when menu is clicked', () => {
    const menu = wrapper.find(FontAwesomeIcon);
    const click = menu.simulate('click');
    expect(click.prop('icon')).toBe(faTimes);
    // const onClick = jest.fn();
    // const component = mount(<FontAwesomeIcon onClick={onClick} />);
    // const result = component.simulate("click");
    // expect(result.instance().props.children.props.faBars).toBe(true);
    // expect(menu.instance().props.icon).toBe(faBars);
    // expect(menu.props().children.icon).toBe(faBars);
  });

  // it("should update state on click", () => {
  //   const onClick = jest.fn();
  //   const wrapper = mount(<FontAwesomeIcon onClick={onClick} />);
  //   const handleClick = jest.spyOn(React, "useState");
  //   handleClick.mockImplementation((IsNavOpen) => [isNavOpen, setIsNavOpen]);
  //   wrapper.find(FontAwesomeIcon).simulate("click");
  //   expect(setIsNavOpen).toBe(true);
  // });

  it('should change the display property from static to none based on props', () => {
    const holder = shallow(<Links />);
    expect(holder).toMatchSnapshot();
  });
});
