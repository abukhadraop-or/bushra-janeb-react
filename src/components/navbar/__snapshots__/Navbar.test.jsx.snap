// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should change the display property from static to none based on props 1`] = `ShallowWrapper {}`;
